File: app/src/main/java/com/example/simplehabitassignment/network/DataAgent.java
package com.example.simplehabitassignment.network;


import com.example.simplehabitassignment.delegates.CategoriesAndProgramsResponseDelegate;
import com.example.simplehabitassignment.delegates.CurrentProgramResponseDelegate;
import com.example.simplehabitassignment.delegates.TopicsResponseDelegate;

public interface DataAgent {


    void getTopics(String accessToken, int page, TopicsResponseDelegate delegate);
    void getCategoriesAndPrograms(String accessToken, int page, CategoriesAndProgramsResponseDelegate delegate);
    void getCurrentProgram(String accessToken, int page, CurrentProgramResponseDelegate delegate);
}

File: app/src/main/java/com/example/simplehabitassignment/network/DataAgent.kt
package com.example.simplehabitassignment.network


import com.example.simplehabitassignment.delegates.CategoriesAndProgramsResponseDelegate
import com.example.simplehabitassignment.delegates.CurrentProgramResponseDelegate
import com.example.simplehabitassignment.delegates.LoginDelegate
import com.example.simplehabitassignment.delegates.TopicsResponseDelegate

interface DataAgent {


    fun getTopics(accessToken: String, page: Int, delegate: TopicsResponseDelegate)
    fun getCategoriesAndPrograms(accessToken: String, page: Int, delegate: CategoriesAndProgramsResponseDelegate)
    fun getCurrentProgram(accessToken: String, page: Int, delegate: CurrentProgramResponseDelegate)
    fun login(phone: String, password: String, loginDelegate: LoginDelegate)
}
